posthog/api/special_migration.py

```diff
@@ -0,0 +1,100 @@
+            "current_query_id",
+            "celery_task_id",
+            "started_at",
+            "finished_at",
+            "last_error",
+            "posthog_max_version",
+            "posthog_min_version",
+        ]
+        read_only_fields = [
+            "id",
+            "name",
+            "description",
+            "progress",
+            "status",
+            "current_operation_index",
+            "current_query_id",
+            "celery_task_id",
+            "started_at",
+            "finished_at",
+            "last_error",
+            "posthog_max_version",
+            "posthog_min_version",
+        ]
+
+
+class SpecialMigrationsViewset(StructuredViewSetMixin, viewsets.ModelViewSet):
+    queryset = SpecialMigration.objects.all()
+    permission_classes = [StaffUser]
+    serializer_class = SpecialMigrationSerializer
+
+    @action(methods=["POST"], detail=True)
+    def trigger(self, request, **kwargs):
+        if len(get_all_running_special_migrations()) >= MAX_CONCURRENT_SPECIAL_MIGRATIONS:
+            return response.Response(
+                {
+                    "success": False,
+                    "error": f"No more than {MAX_CONCURRENT_SPECIAL_MIGRATIONS} special migration can run at once.",
+                },
+                status=400,
+            )
+
+        migration_instance = self.get_object()
+
+        if not is_migration_in_range(migration_instance.posthog_min_version, migration_instance.posthog_max_version):
+            return response.Response(
+                {
+                    "success": False,
+                    "error": f"Can't run migration. Minimum PostHog version: {migration_instance.posthog_min_version}. Maximum PostHog version: {migration_instance.posthog_max_version}",
+                },
+                status=400,
+            )
+
+        migration_instance.status = MigrationStatus.Starting
+        migration_instance.save()
+
+        trigger_migration(migration_instance)
+        return response.Response({"success": True}, status=200)
+
+    # DANGEROUS! Can cause another task to be lost
+    @action(methods=["POST"], detail=True)
+    def force_stop(self, request, **kwargs):
+        migration_instance = self.get_object()
+        if migration_instance.status != MigrationStatus.Running:
+            return response.Response(
+                {"success": False, "error": "Can't stop a migration that isn't running.",}, status=400,
+            )
+        force_stop_migration(migration_instance)
+        return response.Response({"success": True}, status=200)
+
+    @action(methods=["POST"], detail=True)
+    def force_rollback(self, request, **kwargs):
+        migration_instance = self.get_object()
+        if migration_instance.status != MigrationStatus.CompletedSuccessfully:
+            return response.Response(
+                {"success": False, "error": "Can't force rollback a migration that did not complete successfully.",},
```

Comment:
why can't we `force` commonly means do it regardless of state, same as above, if we want them to check things they shouldn't be `force_` prefixed

Reply:
Yeah - this seems like another where I would expect the function to roll back the migration no matter if it was fully or partially applied.

Reply:
Agreed, will rename and update the logic